src/BSON/ObjectId.c

```diff
@@ -336,6 +336,32 @@ static phongo_create_object_retval php_phongo_objectid_create_object(zend_class_
 #endif
 } /* }}} */
 
+static phongo_create_object_retval php_phongo_objectid_clone_object(zval* object TSRMLS_DC) /* {{{ */
+{
+	php_phongo_objectid_t* intern;
+	php_phongo_objectid_t* new_intern;
+	phongo_create_object_retval new_object;
+
+	intern = Z_OBJECTID_OBJ_P(object);
+	new_object = php_phongo_objectid_create_object(Z_OBJCE_P(object) TSRMLS_CC);
+
+#if PHP_VERSION_ID >= 70000
+	new_intern = Z_OBJ_OBJECTID(new_object);
+	zend_objects_clone_members(&new_intern->std, &intern->std TSRMLS_CC);
+#else
+	{
+		zend_object_handle handle = Z_OBJ_HANDLE_P(object);
+
+		new_intern = (php_phongo_objectid_t*) zend_object_store_get_object_by_handle(new_object.handle TSRMLS_CC);
+		zend_objects_clone_members(&new_intern->std, new_object, &intern->std, handle TSRMLS_CC);
+	}
+#endif
+
+	php_phongo_objectid_init_from_hex_string(new_intern, intern->oid, 24 TSRMLS_CC);
```

Comment:
Would it make sense to also use `memcpy()` here as you did for Decimal128? That would avoid the string conversion in `php_phongo_objectid_init_from_hex_string`.

Reply:
Good point 👍